test/new-e2e/tests/npm/ec2_1host_containerized_test.go

```diff
@@ -0,0 +1,182 @@
+	v.BaseSuite.BeforeTest(suiteName, testName)
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.BaseSuite.IsDevMode() {
+		v.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	}
+}
+
+// TestFakeIntakeNPMHostRequests Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+// 2 tests generate the request on the host and on docker
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMContainerizedSuite) TestFakeIntakeNPM_HostRequests() {
+	testURL := "http://" + v.Env().HTTPBinHost.Address + "/"
+
+	// generate a connection
+	v.Env().RemoteHost.MustExecute("curl " + testURL)
+
+	test1HostFakeIntakeNPM(&v.BaseSuite, v.Env().FakeIntake)
+}
+
+// TestFakeIntakeNPMDockerRequests Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+// 2 tests generate the request on the host and on docker
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMContainerizedSuite) TestFakeIntakeNPM_DockerRequests() {
+	testURL := "http://" + v.Env().HTTPBinHost.Address + "/"
+
+	// generate a connection
+	v.Env().RemoteHost.MustExecute("docker run public.ecr.aws/k8m1l3p1/alpine/curler curl " + testURL)
+
+	test1HostFakeIntakeNPM(&v.BaseSuite, v.Env().FakeIntake)
+}
+
+// TestFakeIntakeNPM600cnxBucket_HostRequests Validate the agent can communicate with the (fake) backend and send connections
+// every 30 seconds with a maximum of 600 connections per payloads, if more another payload will follow.
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for n payloads and check if the last 2 have a maximum span of 100ms
+func (v *ec2VMContainerizedSuite) TestFakeIntakeNPM600cnxBucket_HostRequests() {
+	testURL := "http://" + v.Env().HTTPBinHost.Address + "/"
+
+	// generate connections
+	v.Env().RemoteHost.MustExecute("ab -n 600 -c 600 " + testURL)
+
+	test1HostFakeIntakeNPM600cnxBucket(&v.BaseSuite, v.Env().FakeIntake)
+}
+
+// TestFakeIntakeNPM600cnxBucket_DockerRequests Validate the agent can communicate with the (fake) backend and send connections
+// every 30 seconds with a maximum of 600 connections per payloads, if more another payload will follow.
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for n payloads and check if the last 2 have a maximum span of 100ms
+func (v *ec2VMContainerizedSuite) TestFakeIntakeNPM600cnxBucket_DockerRequests() {
+	testURL := "http://" + v.Env().HTTPBinHost.Address + "/"
+
+	// generate connections
+	v.Env().RemoteHost.MustExecute("docker run public.ecr.aws/docker/library/httpd ab -n 600 -c 600 " + testURL)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
This probably can run to the docker manager and called from the docker client. Can be addressed in a follow up.